### ISSUE TITLE
Ingest Node preprocessor

### ISSUE THREAD

**Issue by charles-pinkston** [codelibs/fess#1402]
Is there any way to add an ingest note pipeline?  I have some custom metadata fields that Fess is reading in properly, but I need to be able to split them into an array.

**Comment by marevol** [comment]
Need more info...

**Comment by puecher** [comment]
It would be great to split the "content" field into an array (new keyword datatype field called "content.keywords"). Any hints?

**Comment by marevol** [comment]
You can edit app/WEB-INF/classes/fess_indices/fess/doc.json.
fess index is created at first time startup.

**Comment by puecher** [comment]
Thank you for the info @marevol. How do I split the "content" field by single space (String[] keywords = content.split(" ")) and store the values in the new "content.keywords" field?

I would like to get a term cloud per index/alias (Term Vectors unfortunately support only 1+ documents).

**Comment by charles-pinkston** [comment]
In my case, I had a bunch of values that were separated by a '|' character in a customized field.

In order to make it work, I added a new pattern tokenizer, and a custom analyzer to use that in the fess.json file.  Then in my doc.json, I changed my field to type text, fielddata true, and used the new analyzer.  Not sure if that would work for you, but it might give you a couple of other ideas to work with.
